Add Settings On Title Screen, Update Starter-Select Tutorial Dialogue (EN, DE, FR, CN, BR)

torranx opened the pull request:
Resolves #1312 

## [PR changes from initial review](https://github.com/pagefaultgames/pokerogue/pull/1367#issuecomment-2131658731)

## What are the changes?
- Added Menu option on the title phase*
- Changed starter select dialogue**
- Removed the access-menu tutorial when entering the encounter phase***

## Why am I doing these changes?
If not all, most games have the option to access settings before playing. While it's true that we can access the settings immediately when first playing up PokeRogue, _it is not apparent how exactly you can access it_ until you start a new game and the tutorial to open it appears. 

- *(Accessibility) Having the menu option prominently displayed on the title screen ensures that players can easily see and access it without needing to guess or remember a specific button press. This is particularly helpful for new players who may not be familiar with all the controls yet.
- *(Accessibility and Standardization) Users typically expect to find key options like settings, load game, and other menu functions on the title screen. Placing these options where they are expected improves the user experience by aligning with common interface design practices. 
- ** It is not apparent that you have to press the Space bar to select a Pokemon, this is based from my experience and others as well.
- *** Adding the Menu on the title screen makes the current tutorial encounter phase flow obsolete.


## What did change?
- New option (Menu) in title screen
- Starter select first line changed from `From this screen, you can select your starters` to `From this screen, you can select your starters by pressing Z or the Space bar`
- The access-menu tutorial found on the encounter phase is moved to the end of menu tutorial, phrased like so `You can also access menu by pressing M or Escape anytime while awaiting input.`

### Screenshots/Videos
- Menu option in title screen
<img width="1251" alt="Screenshot 2024-05-25 at 9 27 07 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/050baf2e-a290-4dfa-856c-bfb12fe7222e">


- New line on menu tutorial 


https://github.com/pagefaultgames/pokerogue/assets/68144167/fcad40fe-9dd8-4aea-9f77-582baad1925b



- Starter select tutorial update
<img width="1237" alt="Screenshot 2024-05-25 at 9 27 22 PM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/cd255350-ccff-4b92-9120-fb160e6f24ad">


- Removed the access-menu tutorial on encounter phase

https://github.com/pagefaultgames/pokerogue/assets/68144167/8a62e519-20b1-4473-b7c1-53db869c758d



## How to test the changes?
- Open up the app on incognito follow the tutorial up to encounter phase.

## Checklist
- [x] Have I tested the changes (manually)?
- [x] Are the changes visual?
- [x] Have I made sure the PR is self-contained and cannot be split into smaller PRs? (Localization can be done after/if merged)

bennybroseph commented:
I think this is fine, but Menu should be the final option, not in the middle like that.

torranx commented:
> I think this is fine, but Menu should be the final option, not in the middle like that. 
I see, I decided to place it before the Daily Run because of the (Beta) label

> You can also use Z to select starters, not just Space Bar
Thanks for pointing this out

Will update the PR

CodeTappert commented on src/locales/de/menu.ts:
```diff
@@ -11,6 +11,7 @@ export const menu: SimpleTranslationEntries = {
   "dailyRun": "Täglicher Run (Beta)",
   "loadGame": "Spiel laden",
   "newGame": "Neues Spiel",
+  "menu": "Menu",
```
"Menü"

torranx replied:
added, thanks!

CodeTappert commented on src/locales/de/tutorial.ts:
```diff
@@ -8,11 +8,10 @@ export const tutorial: SimpleTranslationEntries = {
     $Dort kannst du u. A. die Spielgeschwin-\ndigkeit und das Fensterdesign ändern.
-    $Das Menü verbirgt noch andere Funktionen - probier' sie gerne aus!`,
+    $Das Menü verbirgt noch andere Funktionen - probier' sie gerne aus!
+    $You can also access menu by pressing M or Escape anytime while awaiting input.`,
```
$Das Menü kann jederzeit durch einen Druck auf M oder Esc aufgerufen werden`,

torranx replied:
added, thanks!

torranx commented:
For those interested to help on the localization, here are the ff text needed:
- Menu
- From this screen, you can select your starters by pressing Z or the Space bar.
- You can also access menu by pressing M or Escape anytime while awaiting input.

molker commented:
Really appreciate you adding all the locales for this.
I think the one change I'd like before reviewing further: 

- I agree that users tend to expect a way to access the menu from the title but since there is a tutorial outlining how to get the menu, I think it might be better to change "Menu" to "Settings" and go straight to the settings menu instead of popping up the menu

torranx commented:
> * there is a tutorial outlining how to get the menu, I think it might be better to change "Menu" to "Settings" and go straight to the settings menu instead of popping up the menu

The PR includes the removal of that tutorial, instead I added it as the last sentence to the tutorial when opening up Menu. I'd have to bring that back if that is what we want.

molker commented:
> > * there is a tutorial outlining how to get the menu, I think it might be better to change "Menu" to "Settings" and go straight to the settings menu instead of popping up the menu
> 
> The PR includes the removal of that tutorial, instead I added it as the last sentence to the tutorial when opening up Menu. I'd have to bring that back if that is what we want.

I think its fine to put it back, I'd like to soon add a controller menu to settings anyway so people will be able to see the options there

torranx commented:
Updated PR with the ff:
- Added back encounter phase access-menu tutorial
- Removed "You can also access menu by pressing M or Escape anytime while awaiting input." from Menu tutorial.
- Replaced Menu with Settings

CodeTappert commented:
> For those interested to help on the localization, here are the ff text needed:
> 
> * Settings
> * From this screen, you can select your starters by pressing Z or the Space bar.

DE:
- Settings: "Einstellungen"
- From this screen, you can select your starters by pressing Z or the Space bar.: "In diesem Bildschirm kannst du mit Z oder Leertaste deine Starter auswählen"

Adri1 commented:
PR subbmited to your branch with French updates

molker commented:
@torranx Looks like its covering the text for how many active players, if you fix that I can finish my review and approve

torranx commented:
> @torranx Looks like its covering the text for how many active players, if you fix that I can finish my review and approve

Nice catch! 
Screenshots for reference:
<img width="1300" alt="Screenshot 2024-05-28 at 8 45 13 AM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/fb721710-b699-4536-ae7a-f161063cecfc">

<img width="1300" alt="Screenshot 2024-05-28 at 8 46 17 AM" src="https://github.com/pagefaultgames/pokerogue/assets/68144167/76b3013e-dc02-4427-89d4-e5970565e7cc">

SnowCharmQ commented:
Thanks for your work on this PR to solve my ISSUE :)
This is pretty helpful for those who are new to PokeRogue including me.

SnowCharmQ commented:
> For those interested to help on the localization, here are the ff text needed:
> 
> * Settings
> * From this screen, you can select your starters by pressing Z or the Space bar.

zh_CN:
- Settings: "设置"
- From this screen, you can select your starters by pressing Z or the Space bar.: "在当前页面，你可以通过按Z或空格键来选择你的初始宝可梦"

Thanks :)

josericardo-fo commented:
pt_BR:

Settings: "Configurações"
From this screen, you can select your starters by pressing Z or the Space bar.: "Aqui você pode escolher seus iniciais apertando a tecla Z ou a Barra de Espaço."
